### ISSUE TITLE
Fix `*-case|*-unknown` false positives for CSS-in-JS template literals

### ISSUE THREAD

**Issue by Nume** [stylelint/stylelint#6659]
### What steps are needed to reproduce the bug?

It seems that rule `function-no-unknown` is triggering a false positive when functions are called within an anonymous function

```
import styled from "styled-components"

function getColor() {
  return "red"
}

const Foo = styled.div`
  color: ${() => getColor()}; // Unexpected unknown function "getColor"
`

const Bar = styled.div`
  color: ${getColor()}; // OK
`
```

### What Stylelint configuration is needed to reproduce the bug?

```
module.exports = {
  extends: [
    "stylelint-config-standard",
  ],
  rules: {},
  customSyntax: "postcss-styled-syntax",
}
```

### How did you run Stylelint?

`stylelint \"./src/**/*.tsx\"`

### Which version of Stylelint are you using?

`15.1.0`

```
"postcss-styled-syntax": "0.3.1",
"stylelint": "15.1.0",
"stylelint-config-standard": "30.0.1"
```

### What did you expect to happen?

It should not report `Unexpected unknown function "getColor"` since we are inside a literal expression

### What actually happened?

```
✖  Unexpected unknown function "getColor"  function-no-unknown
```

### Does the bug relate to non-standard syntax?

yes, css-in-js

**Comment by jeddy3** [comment]
@Nume Thanks for the report and for using the template.

Please raise an issue in the [`postcss-styled-syntax` tracker](https://github.com/hudochenkov/postcss-styled-syntax/issues) as custom syntaxes are responsible for extracting CSS out of the container and passing it along to Stylelint's built-in rules.

Closing here as related to a community custom syntax rather than Stylelint itself.

**Comment by Nume** [comment]
It was previously opened at postcss-styled-syntax https://github.com/hudochenkov/postcss-styled-syntax/issues/6 but closed as related to bug in rule and not the custom syntax :)

**Comment by jeddy3** [comment]
@Nume Thanks for the update.

As requested in https://github.com/hudochenkov/postcss-styled-syntax/issues/6#issuecomment-1428343647, I'll mentioned @hudochenkov to take a look.

**Comment by hudochenkov** [comment]
Identified more rules in my work project, which shows similar symptom: trying to lint non-standard CSS declaration value.

- `unit-no-unknown`
- `value-keyword-case`
- `annotation-no-unknown`

---

```js
const ListStyled = styled.div`
    padding: ${(props) => (props.isContained ? `0 ${token.spacing5}` : `${token.spacing3} 0`)};
`;
```

```
src/components/List/List.tsx
 2:89  ✖  Unexpected unknown unit "`"  unit-no-unknown
```

---

```js
const BadgeStyled = styled.div`
    height: ${(props) => (props.isSmall ? `20px` : `24px`)};
`;
```

```
src/components/Badge/Badge.tsx
   2:24  ⚠  Expected "props.isSmall" to be "props.issmall"            value-keyword-case
```

---

```js
const Container = styled.label`
    cursor: ${(props) => !props.isDisabled && 'pointer'};
`;
```

```
src/components/Checkbox/Checkbox.tsx
 2:23  ✖  Unexpected unknown annotation "!props.isDisabled"  annotation-no-unknown
```

**Comment by alexkuz** [comment]
@hudochenkov there is the same kind of issue with `media-feature-name-no-unknown`:

```
  @media (${(p) => p.theme.breakpoints.netbook}) {
    position: absolute;
```
```
42:14  ✖  Unexpected unknown media feature name "p"  media-feature-name-no-unknown
```

**Comment by jeddy3** [comment]
@alexkuz Please create [a new issue](https://github.com/stylelint/stylelint/issues/new?assignees=&labels=&template=REPORT_A_BUG.yml) so that we can track the bug.

**Comment by Mouvedia** [comment]
> there is the same kind of issue with `media-feature-name-no-unknown`:
> 
> ```
>   @media (${(p) => p.theme.breakpoints.netbook}) {
>     position: absolute;
> ```
> 
> ```
> 42:14  ✖  Unexpected unknown media feature name "p"  media-feature-name-no-unknown
> ```

This was fixed by version [15.6.2](https://github.com/stylelint/stylelint/releases/tag/15.6.2).
